microsoft/testsuites/core/cpu.py

```diff
@@ -31,8 +31,9 @@ class CPU(testsuite.TestSuite):
     def _set_cpu_state(self, cpu_id: str, state: str, node: Node) -> bool:
         file_path = f"/sys/devices/system/cpu/cpu{cpu_id}/online"
         file_exists = node.shell.exists(PurePosixPath(file_path))
-        if file_exists:
-            node.tools[Echo].write_to_file(state, file_path, shell=True, sudo=True)
+        if not file_exists:
+            return False
+        node.tools[Echo].write_to_file(state, file_path, shell=True, sudo=True)
```

Comment:
BTW, the shell shouldn't be a parameter of this method. It should be always true underlying, because the command doesn't work without it.